![collaboration-logo](../IM/Github_image_banner.png)

# **Obtain Comprehensive Information with Assembly Step**

After obtaining high-quality, filtered reads through quality control processes like those performed by `fastp`, the next crucial step in genomic analysis is **genome assembly**. This process involves reconstructing the original genome sequence from the short DNA fragments generated by sequencing technologies. Assembling these reads into contiguous sequences, or **contigs**, is essential for several reasons:

- **Comprehensive Analysis**: A complete genome assembly provides a holistic view of the organism's genetic makeup, facilitating in-depth studies of gene function, regulatory elements, and genomic architecture.

- **Functional Annotation**: Accurate assemblies enable the identification and annotation of genes, operons, and other functional elements, which are vital for understanding biological processes and pathways.

- **Comparative Genomics**: Assembled genomes allow for meaningful comparisons between different strains or species, aiding in the study of evolutionary relationships and the identification of conserved or unique genetic features.

While some analyses can be performed directly on raw reads, assembling them into longer sequences often enhances the accuracy and interpretability of the results.

## Shovill: Streamlined Assembly for Bacterial Genomes

`shovill` is a specialized pipeline designed to assemble bacterial isolate genomes from Illumina paired-end reads. By leveraging the capabilities of the SPAdes assembler and optimizing pre- and post-assembly processes, Shovill offers a more efficient and user-friendly approach to genome assembly. Key features of Shovill include:

- **Speed and Efficiency**: Shovill streamlines the assembly process, reducing computational time without compromising assembly quality.

- **Compatibility**: While it utilizes SPAdes by default, Shovill also supports other assemblers like SKESA, Velvet, and Megahit, providing flexibility based on user preferences and project requirements.

- **Simplified Workflow**: Shovill automates various stages of the assembly pipeline, making it accessible to users with varying levels of bioinformatics expertise.

**Example Usage of Shovill for Paired-End Illumina Data**

To perform a genome assembly using Shovill with paired-end Illumina data, follow these steps:

1. **Install `shovill`**. You can install `shovill` using `conda`:

   ```bash
   conda install -c conda-forge -c bioconda -c defaults shovill
   shovill --check
   ```
2. **Run `shovill` on your `fastp` Output**. Execute the following command:

   ```bash
   shovill --R1 output_R1_filtered.fastq --R2 output_R2_filtered.fastq --outdir shovill_output --depth 100 --cpus 4
   ```
   **Parameters Explained**:

   - `--R1 output_R1_filtered.fastq`: Specifies the input file containing the forward reads (read 1), here we used `fastp` output from previous run (3.2).

   - `--R2 output_R2_filtered.fastq`: Specifies the input file containing the reverse reads (read 2), also use previous `fastp` output (3.2).

   - `--outdir shovill_output`: Designates the directory where `shovill` will store the assembly results.

   - `--depth 100`: Sets the target coverage depth for the assembly. `shovill` will down-sample the reads to this coverage level to optimize assembly performance.

   - `--cpus 4`: Allocates four CPU cores for the assembly process, allowing `shovill` to parallelize tasks and expedite computation.

By following this workflow, you can efficiently assemble high-quality genomes from your Illumina sequencing data, facilitating downstream analyses such as annotation, comparative genomics, and functional studies.

### References

- `shovill` GitHub Repository: [https://github.com/tseemann/shovill](https://github.com/tseemann/shovill)
- `SPAdes` Genome Assembler GitHub Repository: [https://github.com/ablab/spades](https://github.com/ablab/spades)
