![collaboration-logo](../IM/Github_image_banner.png)

# **Evaluation to Ensure Reliability**

After assembling your sequencing reads into contigs or scaffolds, it's crucial to evaluate the quality of the assembly to ensure its reliability for downstream analyses. Assessing assembly quality helps identify potential issues such as fragmentation, misassemblies, or incomplete regions, which can significantly impact the accuracy of subsequent interpretations. Key metrics to consider include:

- **N50 and L50 Values**: N50 represents the length of the shortest contig at 50% of the total assembly length, providing insight into the assembly's contiguity. L50 indicates the number of contigs that together cover half of the genome, offering a complementary perspective on assembly quality.

- **Total Assembly Length**: Comparing the assembled genome size to the expected genome size can reveal potential gaps or redundancies in the assembly.

- **GC Content**: Analyzing the GC content helps detect biases or contamination, as significant deviations from the expected GC percentage may indicate issues.

- **Misassembly Events**: Identifying structural discrepancies, such as relocations, inversions, or translocations, ensures the structural integrity of the assembly.

## Using QUAST for Assembly Quality Assessment

**QUAST (Quality Assessment Tool for Genome Assemblies)** is a widely used tool that provides detailed reports on various assembly metrics, facilitating the evaluation process. To assess your assembly using QUAST, follow these steps:

1. **Install `quast`**. You can install `quast` using `conda`:

   ```bash
   conda install -c bioconda quast
   ```

2. **Run `quast` on `shovill` output **: Execute the following command:

   ```bash
   quast.py shovill_output/contigs.fa -r reference.fasta -o quast_output
   ```

   **Parameters Explained**:

   - `shovill_output/contigs.fa`: Your assembled genome file with `shovill`.

   - `-r reference.fasta`: (Optional) The reference genome file. Including a reference allows QUAST to perform more comprehensive analyses, such as identifying misassemblies.

   - `-o quast_output`: Specifies the directory where QUAST will store the output reports.

By analyzing the QUAST report, you can gain valuable insights into the quality of your assembly, guiding necessary improvements and ensuring the reliability of your genomic analyses.


### References

- `quast` GitHub Repository: [https://github.com/ablab/quast](https://github.com/ablab/quast)
- `quast` manual: [https://quast.sourceforge.net/docs/manual.html](https://quast.sourceforge.net/docs/manual.html)
